src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,230 @@
+  const [tableCellRefs, setTableCellRefs] = useState<Array<React.RefObject<HTMLTableCellElement>>>([]);
+  const { cellOffsets, updateCellOffsets } = useCellOffsets(tableCellRefs);
+
+  // Compute left table padding
+  const table = tableRefObject.current;
+  const tableLeftPadding = table ? parseInt(getComputedStyle(table).paddingLeft) : 0;
+  const tableRightPadding = table ? parseInt(getComputedStyle(table).paddingRight) : 0;
+
+  const { first = 0, last = 0 } = stickyColumns || {};
+  // Calculate the indexes of the last left and right sticky columns, taking into account the selection column
+  const lastLeftStickyColumnIndex = first + (hasSelection ? 1 : 0);
+  const lastRightStickyColumnIndex = columnsLengthWithSelection - 1 - last;
+
+  // Get the width of the first and last sticky columns using the `cellOffsets` state, or use 0 if it's not available
+  const firstStickyColumnsWidth = cellOffsets.first[lastLeftStickyColumnIndex] ?? 0;
+  const lastStickyColumnsWidth = cellOffsets.last[lastRightStickyColumnIndex] ?? 0;
+
+  // Calculate the sum of all sticky columns' widths
+  const totalStickySpace = firstStickyColumnsWidth + lastStickyColumnsWidth;
+
+  useEffect(() => {
+    // Check if there is enough scrollable space for sticky columns to be enabled
+    const hasEnoughScrollableSpace =
+      totalStickySpace + MINIMUM_SCROLLABLE_SPACE + tableLeftPadding + tableRightPadding < (containerWidth ?? 0);
+
+    // Determine if sticky columns should be disabled based on the conditions
+    const shouldDisable = noStickyColumns || !isWrapperScrollable || !hasEnoughScrollableSpace;
+    setShouldDisable(shouldDisable);
+  }, [
+    containerWidth,
+    noStickyColumns,
+    stickyColumns,
+    totalStickySpace,
+    visibleColumnsLength,
+    tableLeftPadding,
+    tableRightPadding,
+    isWrapperScrollable,
+  ]);
+
+  useEffect(() => {
+    // Create new refs for the visible columns and selection column, if present
+    setTableCellRefs(tableCellRefs =>
+      [...new Array(columnsLengthWithSelection)].map(
+        (_: any, i: number) => tableCellRefs[i] || createRef<HTMLTableCellElement>()
+      )
+    );
+  }, [columnsLengthWithSelection]);
+
+  const getStickyStyles = useCallback(
+    (colIndex: number, stickySide?: 'left' | 'right') => {
+      if (!stickySide) {
+        return { sticky: {}, stuck: {} };
+      }
+      const isFirstColumn = colIndex === 0;
+      let stuck = {};
+      if (isFirstColumn && !hasSelection && tableLeftPadding !== 0) {
+        stuck = { paddingLeft: `${tableLeftPadding}px` };
+      }
+      // Determine the offset of the sticky column using the `cellOffsets` state object
+      const offsetKey = stickySide === 'right' ? 'last' : 'first';
+      const stickyColumnOffset = cellOffsets[offsetKey]?.[colIndex + (hasSelection ? 1 : 0)];
+      return {
+        sticky: typeof stickyColumnOffset !== 'undefined' ? { [stickySide]: `${stickyColumnOffset}px` } : {},
```

Comment:
Can this really be undefined? If so, it might indicate an issue in the code and a potential for a race condition.

Throwing an invariant violation might be extreme if we aren't sure that should actually never happen in the complex React lifecycle, but it would probable make sense to log a console warning so that if reported, we know where to look at.